array_api_tests/test_data_type_functions.py

```diff
@@ -174,6 +175,33 @@ def test_iinfo(dtype):
     # TODO: test values
 
 
+if api_version >= "2022.12":
+
+    def atomic_kinds() -> st.SearchStrategy[Union[DataType, str]]:
+        return xps.scalar_dtypes() | st.sampled_from(list(dh.kind_to_dtypes.keys()))
+
+    @given(
+        dtype=xps.scalar_dtypes(),
+        kind=atomic_kinds() | st.lists(atomic_kinds(), min_size=1).map(tuple),
+    )
+    def test_isdtype(dtype, kind):
+        out = xp.isdtype(dtype, kind)
+
+        assert isinstance(out, bool), f"{type(out)=}, but should be bool [isdtype()]"
+        _kinds = kind if isinstance(kind, tuple) else (kind,)
+        expected = False
+        for _kind in _kinds:
+            if isinstance(_kind, str):
+                if dtype in dh.kind_to_dtypes[_kind]:
+                    expected = True
+                    break
+            else:
+                if dtype == _kind:
```

Comment:
This implicitly assumes dtype equality works, meaning if it's broken or incorrect this might not notice. You can see the test I added in my compat PR compare the known value using dtype string names. Maybe it isn't so important here since we also test `dtype.__eq__` anyway.

Reply:
Yeah. I think we're just out of luck if that's the case. Fortunately this would get checked via a sanity check I wrote in `EqualityMapping`

https://github.com/data-apis/array-api-tests/blob/a533680532bf79cfd6aebbe65a202dc458e7ecfd/array_api_tests/dtype_helpers.py#L62-L63